parser/parser.go

```diff
@@ -124,11 +122,9 @@ func (p *parser) Visit(tree antlr.ParseTree) interface{} {
 		return p.VisitCreateStruct(tree.(*gen.CreateStructContext))
 	}
 
-	text := "<<nil>>"
-	if tree != nil {
-		text = tree.GetText()
-	}
-	panic(fmt.Sprintf("unknown parsetree type: '%+v': %+v [%s]", reflect.TypeOf(tree), tree, text))
+	// The code only gets to this point if there is an error produced earlier.
```

Comment:
This doesn't seem accurate.  How about "The code only gets to this point if the tree didn't match one of the above types."?

Reply:
When the node can't be identified, the `tree` value is `nil` due to a prior error. I haven't seen any cases that would lead me to believe otherwise, let me test for the presence of other errors, report one if none had been found, and return an empty expression otherwise.

Reply:
Good, although the line 126 comment is now clearly wrong - there might not be an error recorded at this point.  Maybe just delete the comment - I think the code is clear enough by itself.

Reply:
Oops, updated.